### PR TITLE
React color onChange and onChangeComplete handler method parameters fixed

### PR DESCRIPTION
**Pull request by sagar7993**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<http://casesandberg.github.io/react-color/#api-onChange>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by sagar7993** [comment]
@LKay @markspolakovs @mntdn @nkgrnkgr @0815Strohhut @dnlfrst @ErickTamayo - please review this PR :-)

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
This is change to popular package and would need sign off from exper

**Comment by sagar7993** [comment]
@sheetalkamat Is there amy specific change you'd like me to make? Github is showing me that you have requested changes. Could you be a little bit more specific please

**Comment by sagar7993** [comment]
@sheetalkamat Please don't click on the request changes option unless you have some specific changes that you'd like me to make. If you click on request changes option, it adds the revision needed label to this PR which might lead to other reviewers to think that something is pending from my side.

Please refer to the official website
http://casesandberg.github.io/react-color/#api-onChange

As per the demos, the onChange and onChangeComplete handlers receive 2 arguments.

File: types/react-color/index.d.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export interface ColorResult {
     rgb: RGBColor;
 }
 
-export type ColorChangeHandler = (color: ColorResult) => void;
+export type ColorChangeHandler = (color: ColorResult, $event: React.ChangeEvent<HTMLInputElement>) => void;
```
Comment (peterblazejewicz) [inline types/react-color/index.d.ts:37]: This looks OK, but the authors of this DT seem to stick to dfiferent naming convention:
```suggestion
export type ColorChangeHandler = (color: ColorResult, event: React.ChangeEvent<HTMLInputElement>) => void;
```

File: types/react-color/react-color-tests.tsx [hunk 2]
```diff
@@ -15,8 +15,8 @@ interface CustomProps extends InjectedColorProps {
 }
 
 const CustomComponent: StatelessComponent<CustomProps> = (props: CustomProps) => {
-    function onChange(color: ColorResult) {
-        console.log(color);
+    function onChange(color: ColorResult, $event: React.ChangeEvent<HTMLInputElement>) {
+        console.log(color, $event);
```
Comment (peterblazejewicz) [inline types/react-color/react-color-tests.tsx:19]:
```suggestion
    function onChange(color: ColorResult, event: React.ChangeEvent<HTMLInputElement>) {
        console.log(color, event);
```

**Comment by sagar7993** [comment]
@peterblazejewicz thanks for the review :-) I have pushed thr change as suggested by you. Please have a look.

**Comment by peterblazejewicz** [comment]
/cc @ErickTamayo @nkgrnkgr @markspolakovs can you folks take a look?
(don't bother with commits, the state is now clean and focused for merge and review)

**Review by ErickTamayo — APPROVED** [review]

**Comment by sagar7993** [comment]
Ready to merge

**Comment by alumni** [comment]
> As per the demos, the onChange and onChangeComplete handlers receive 2 arguments.

@sagar7993 The second argument is optional. It should be optional in the type definitions too.

**Comment by sagar7993** [comment]
@alumni From what I understand after having gone through the code, it seems that the second argument is not optional. Could you share the link to the docs/code where it is mentioned that the second argument is optional? I'd be happy to create a new PR if that's the case.

**Comment by sagar7993** [comment]
> This example: http://casesandberg.github.io/react-color/#api-color

In this example, the 2nd parameter is passed to the function, but the handler here is simply not using it. If you change the function like this -

```
handleChangeComplete = (color, event) => {
    this.setState({ background: color.hex });
    console.log(event);
};
```
You will find that the event parameter is indeed present.

**Comment by linxianxi** [comment]
is a problem?   https://github.com/casesandberg/react-color/issues/738

**Pull request merged**